### ISSUE TITLE
This and super binding inside initializers

### ISSUE THREAD

**Issue by jridgewell** [littledan/proposal-new-initialize#6]
```js
class Base {
    x = 1;
    method() { return 2; }
}

class Sub extends Base {
    z = this.x + super.method();
    constructor() {
      const instance = Reflect.construct(Base, [], new.target);
      new.initialize(instance);
      return instance;
    }
}

const instance = new Sub();
console.log(instance.z)
```

Just so that we're thinking about it. The `this` binding should be easy, but I'm not 100% sure how to change the `super` binding? Does it just work?

**Comment by littledan** [comment]
I'm having trouble understanding what the problem is. I think it should just work?

**Comment by jridgewell** [comment]
`super` binding confuses the hell out of me. I can reason about it when I pretend it's `this.constructor.__proto__`, but I don't actually know how it works in the spec.

What is `super` in this case? Or, what is `super` when I pass in some object literal (and not the result of `Reflect.construct`)?

**Comment by ljharb** [comment]
if `this` becomes rebound, then I’d expect `super` to; and if not, not.

If not, I’d expect super.method to be the one on Base.prototype (no matter what instance was)

**Comment by littledan** [comment]
Yes, the whole idea of the feature is to pass in a different this value to initialize.

Returning something from the super constructor is the "normal" way the subclass instantiation process, and constructors can return whatever object they wabt. The return override trick is just what you get when you combine features. We would have to do something special (and weird) to block it in this case.

**Comment by ljharb** [comment]
What I meant was this comment:
```js
class Sub extends Base {
    z = this.x + super.method();
    constructor() {
      const instance = Reflect.construct(Base, [], new.target);
      new.initialize(instance);
      assert(this === instance); // and this would be true?
      return instance; // this line can be removed, and it'd do the same thing?
    }
}
```

**Comment by littledan** [comment]
Oh, no, I am only talking about the this binding within the initializer, not the constructor.

**Comment by ljharb** [comment]
ok so, calling `new.initialize` doesn’t alter the `this` binding, so I’d also expect the `super` binding and the method HomeObjects to be unchanged.

**Comment by ljharb** [comment]
@littledan so `super` in initializers becomes what, exactly? Would it be the initial value of `replacement.__proto__`?

**Comment by littledan** [comment]
See the detailed syntax and semantics described [in this section of the explainer](https://github.com/littledan/proposal-new-initialize#detailed-syntax-and-semantics). I'm not proposing any changes to the semantics of super.

In particular, `new.initialize` doesn't set the prototype, it just initializes fields and private methods.
